### PR TITLE
Add paste buttons

### LINKED ISSUE #16: Delete files and tell user to use paste.ubuntu.com instead
due to recent security concerns with downloading files, it might be advisable for absol to delete them and tell the sender to use https://paste.ubuntu.com

### PR DESCRIPTION
**Pull request by randomairborne**
This requires a [Pastebin](https://github.com/randomairborne/mcpaste) that Absol's link button points to to be useful.

Closes #16.

### CODE REVIEW

File: modules/pastes/pastes.go [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+		logger.Err().Fatal("At least one guild ID is required to use pastes module!")
+	}
+}
+
+func HandleMessage(ds *discordgo.Session, mc *discordgo.MessageCreate) {
+	if len(mc.Attachments) <= 0 {
+		return
+	}
+	used := false
+	for _, item := range strings.Split(viper.GetString("paste.guilds"), " ") {
```
Comment (LordRalex) [inline modules/pastes/pastes.go]: We use ; as the delimiter for ENV settings

File: modules/pastes/pastes.go [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+	if !used {
+		return
+	}
+	rows := []discordgo.MessageComponent{}
+	row := []discordgo.MessageComponent{}
+	for idx, element := range mc.Attachments {
+		if element.ContentType == "text/plain; charset=utf-8" || element.ContentType == "application/json; charset=utf-8" || element.ContentType == "text/html; charset=utf-8" {
+			btn := discordgo.Button{
+				Emoji: discordgo.ComponentEmoji{
+					Name: "📜",
```
Comment (LordRalex) [inline modules/pastes/pastes.go:47]: Please use a hardcoded emoji id or something here

File: modules/pastes/pastes.go [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+				},
+				Label: "View " + element.Filename,
+				Style: discordgo.LinkButton,
+				URL:   viper.GetString("paste.url") + mc.ID + "/" + element.Filename,
```
Comment (LordRalex) [inline modules/pastes/pastes.go]: Assume the url doesn't have a / at it's start

File: modules/pastes/pastes.go [hunk 4]
```diff
@@ -0,0 +1,72 @@
...
+			}
+			row = append(row, btn)
+			if (len(row) >= 5 || idx+1 == len(mc.Attachments)) && len(row) > 0 {
```
Comment (LordRalex) [inline modules/pastes/pastes.go]: Just handle >5 here, move the "always add the row to rows" to after the loop.

File: modules/pastes/pastes.go [hunk 5]
```diff
@@ -0,0 +1,72 @@
...
+				rows = append(rows, discordgo.ActionsRow{Components: row})
+				row = []discordgo.MessageComponent{}
+			}
+		}
+	}
+	if len(rows) <= 0 {
+		return
+	}
+	msg := &discordgo.MessageSend{
```
Comment (LordRalex) [inline modules/pastes/pastes.go:66]: Probably should use this as a reply, so that we can see what message it's related to

**Review by LordRalex — CHANGES_REQUESTED** [review]

File: modules/pastes/pastes.go [hunk 6]
```diff
@@ -47,22 +47,26 @@ func HandleMessage(ds *discordgo.Session, mc *discordgo.MessageCreate) {
 				},
 				Label: "View " + element.Filename,
 				Style: discordgo.LinkButton,
-				URL:   viper.GetString("paste.url") + mc.ID + "/" + element.Filename,
+				URL:   viper.GetString("paste.url") + '/' + mc.Channel.ID + '/' + mc.ID + '/' + element.Filename,
```
Comment (LordRalex) [inline modules/pastes/pastes.go]: Recommend using fmt.Sprintf for this btw

**Comment by randomairborne** [comment]
Discord builtin emojis don't have IDs, that's the only possible way to do it with an emoji that comes with discord

**Pull request merged**